packages/embed-components/src/Theme/state/sagas.ts

```diff
@@ -0,0 +1,132 @@
+ SOFTWARE.
+
+ */
+import { takeEvery, put, call } from 'typed-redux-saga'
+import type { PayloadAction } from '@reduxjs/toolkit'
+import type { ITheme } from '@looker/sdk'
+import {
+  registerThemeService,
+  getThemeService,
+  EmbedUrl,
+} from '@looker/embed-services'
+import { themeActions } from './slice'
+import type { SelectThemeAction } from './slice'
+
+/**
+ * Registers theme service with factory
+ */
+function* initSaga() {
+  const { initSuccessAction, setFailureAction } = themeActions
+  try {
+    registerThemeService()
+    yield* put(initSuccessAction())
+  } catch (error: any) {
+    yield* put(setFailureAction({ error: error.message }))
+  }
+}
+
+/**
+ * Gets all themes, default and url themes, and sets the selected theme
+ */
+function* loadThemeDataSaga() {
+  const { loadThemeDataSuccessAction, setFailureAction } = themeActions
+  try {
+    const service = getThemeService()
+    yield* call([service, 'load'])
+    const searchParams = new EmbedUrl().searchParams
+    const urlThemeName = searchParams.theme
+    let urlTheme: ITheme | undefined
+    if (urlThemeName) {
+      urlTheme = service.find('name', urlThemeName)
+    }
+    yield* put(
+      loadThemeDataSuccessAction({
+        themes: service.items,
+        defaultTheme: service.defaultTheme!,
+        selectedTheme: (urlTheme ?? service.defaultTheme)!,
+      })
+    )
+  } catch (error: any) {
+    yield* put(setFailureAction({ error: error.message }))
+  }
+}
+
+/**
+ * Gets all available themes
+ */
+function* getThemesSaga() {
+  const { getThemesSuccessAction, setFailureAction } = themeActions
+  try {
+    const service = getThemeService()
+    yield* call([service, 'getAll'])
+    yield* put(getThemesSuccessAction({ themes: service.items }))
+  } catch (error: any) {
+    yield* put(setFailureAction({ error: error.message }))
+  }
+}
+
+/**
+ * Gets default theme
+ */
+function* getDefaultThemeSaga() {
+  const { getDefaultThemeSuccessAction, setFailureAction } = themeActions
+  try {
+    const service = getThemeService()
+    const defaultTheme = yield* call([service, 'getDefaultTheme'])
+    yield* put(getDefaultThemeSuccessAction({ defaultTheme }))
+  } catch (error: any) {
+    yield* put(setFailureAction({ error: error.message }))
+  }
+}
+
+/**
+ * Sets the selected theme by id
+ * @param action containing id of theme to select
+ */
+function* selectThemeSaga(action: PayloadAction<SelectThemeAction>) {
+  const { selectThemeSuccessAction, setFailureAction } = themeActions
+  try {
+    const service = getThemeService()
+    const selectedTheme = yield* call([service, 'get'], action.payload.id)
+    yield* put(selectThemeSuccessAction({ selectedTheme }))
+  } catch (error: any) {
+    yield* put(setFailureAction({ error: error.message }))
+  }
+}
+
+export function* saga() {
+  const {
+    initAction,
+    loadThemeDataAction,
+    getThemesAction,
+    getDefaultThemeAction,
+    selectThemeAction,
+  } = themeActions
```

Comment:
The actions can be initialized outside of the function

Reply:
I couldn't get this to work. I suspect it has to do with the version of @looker/redux that we are using here.